### ISSUE TITLE
Support for Buffer & ReadableStream data inputs

### ISSUE THREAD

**Issue by austincondict-AngelOak** [k-yle/pdf-to-img#21]
Hi @k-yle, 

I've recently come across this package while looking for a PDF-to-PNG utility and it's been pretty convenient for my purposes. However, I see that the PDF input type is required to be a string representing a URL or local file path. 

My application deals mostly with streamed & buffered PDF contents and I was curious if you'd be interested in adding support for passing the input PDF data as a Buffer or ReadableStream?

I have implemented & tested this locally in my installed node_modules, so if you're open to this enhancement but don't feel up to making the change then I'd be happy to submit a PR, just let me know.

**Comment by k-yle** [comment]
Hi @austincondict-AngelOak feel free to submit a PR if you've already got it working :) 

Otherwise, if you can wait until sunday, I can add this myself

**Comment by austincondict-AngelOak** [comment]
Thanks for the quick reply! I can wait until Sunday, and it may be best for you to make the changes after all as I'm probably too unfamiliar w/ typescript & browser compatibility caveats to confidently contribute a solid patch (never actually work in a browser myself, strictly backend Node v14).

If it may help to save you some time, here's the only thing I tweaked in my index.js -- for the `data` assignment at the top of the `pdf` function:

```javascript
	let data;
	if (typeof pathOrDataUrl === 'string') {
		if (pathOrDataUrl.startsWith(PREFIX)) {
			data = Buffer.from(pathOrDataUrl.slice(PREFIX.length), "base64");  // Received URL
		}
		else {
			data = new Uint8Array(yield fs_1.promises.readFile(pathOrDataUrl));  // Received file path
		}
	}
	else if (typeof pathOrDataUrl === 'object' && pathOrDataUrl != null) {
		if (Buffer.isBuffer(pathOrDataUrl)) {
			data = pathOrDataUrl;  // Received Buffer
		}
		else if (typeof pathOrDataUrl._read === 'function' && typeof pathOrDataUrl._readableState === 'object') {
			data = yield streamToBuffer(pathOrDataUrl);  // Received ReadableStream
		}
	}
```

where `streamToBuffer` looks like:

```javascript
async function streamToBuffer(readableStream)
{
	const chunks = [];
	for await (let chunk of readableStream) {
		chunks.push(chunk);
	}
	return Buffer.concat(chunks);
}
```

**Comment by k-yle** [comment]
@austincondict-AngelOak thanks for the code snippet, that was really helpful! I've made the change an released it in v1.1.0

let me know if you've got any questions
